ngraph/test/util/unary_test.hpp

```diff
@@ -0,0 +1,293 @@
+            }
+
+            // caller have to pass two level of braces when aggregate-initialize input argument of
+            // type std::vector<T>. by adding an overload of type std::initializer_list<T>, we can
+            // save one level of braces
+            Data(const std::initializer_list<T>& v)
+                : value(v.size())
+                , shape{}
+                , no_shape(true)
+            {
+                std::copy(v.begin(), v.end(), value.begin());
+            }
+
+            // we would duplicate by circular-copy or truncate initializer list when number of
+            // element provided in initializer list is not matching the desired shape
+            Data(const std::initializer_list<T>& v, const ngraph::Shape& s)
+                : value(ngraph::shape_size(s))
+                , shape{s}
+            {
+                auto it = v.begin();
+                for (size_t i = 0; i < value.size(); ++i, ++it)
+                {
+                    if (it == v.end())
+                        it = v.begin();
+                    value[i] = *it;
+                }
+            }
+
+        private:
+            std::vector<T> value{};
+            ngraph::Shape shape{};
+            bool no_shape = false;
+
+            template <typename TestEngine, ngraph::element::Type_t et>
+            friend class unary_test;
+        };
+
+        template <typename TestEngine, ngraph::element::Type_t et>
+        class unary_test
+        {
+        public:
+            unary_test(std::shared_ptr<ngraph::Function> function)
+                : m_function(function)
+            {
+            }
+
+            template <ngraph::element::Type_t eleType = et>
+            void test(const std::initializer_list<Data<eleType>>& inputs,
+                      const std::initializer_list<Data<eleType>>& expeted,
+                      Tolerance tol = {})
+            {
+                if (m_function->is_dynamic())
+                {
+                    auto test_case =
+                        ngraph::test::TestCase<TestEngine, ngraph::test::TestCaseType::DYNAMIC>(
+                            m_function);
+                    do_test<eleType>(test_case, inputs, expeted, tol);
+                }
+                else
+                {
+                    auto test_case =
+                        ngraph::test::TestCase<TestEngine, ngraph::test::TestCaseType::STATIC>(
+                            m_function);
+                    do_test<eleType>(test_case, inputs, expeted, tol);
+                }
+            }
+
+            // SISO: single inputs, single output
+            template <ngraph::element::Type_t eleType = et>
+            void test(const Data<eleType>& input, const Data<eleType>& expeted, Tolerance tol = {})
+            {
+                test<eleType>(
+                    std::initializer_list<Data<eleType>>{input}, std::initializer_list<Data<eleType>>{expeted}, tol);
```

Comment:
Why we need to create explicit `initializer_list`?
```suggestion
                test<eleType>({input}, {expeted}, tol);
```

Reply:
Good idea, Changed.